malariagen_data/ag3.py

```diff
@@ -1817,17 +2035,27 @@ def gene_cnv_frequencies(
 
         """
 
+        # resolve region
+        region = self._resolve_region(region)
+
         # get gene copy number data
-        ds_cnv = self.gene_cnv(contig=contig, sample_sets=sample_sets)
+        ds_cnv = self.gene_cnv(region=region, sample_sets=sample_sets)
 
         # get sample metadata
         df_meta = self.sample_metadata(sample_sets=sample_sets)
 
-        # get genes
-        df_genes = self.geneset().query(f"type == 'gene' and contig == '{contig}'")
+        # find genes in the region
+        if region.start and region.end:
+            df_genes = self.geneset().query(
+                f"type == 'gene' and contig == '{region.contig}' and start >= {region.start} and end <= {region.end}"
```

Comment:
```suggestion
                f"type == 'gene' and contig == '{region.contig}' and start <= {region.end} and end >= {region.start}"
```

Possible tweak to retrieve all genes overlapping the requested region.